pkg/gadgets/run/types/metadata.go

```diff
@@ -325,28 +364,41 @@ func (m *GadgetMetadata) populateTracers(spec *ebpf.CollectionSpec) error {
 	return nil
 }
 
-// getGadgetIdentByPrefix returns the string generated by GADGET_ macros.
-func getGadgetIdentByPrefix(spec *ebpf.CollectionSpec, prefix string) string {
+// getGadgetIdentByPrefix returns the strings generated by GADGET_ macros.
+func getGadgetIdentByPrefix(spec *ebpf.CollectionSpec, prefix string) ([]string, error) {
+	var resultNames []string
+	var resultError error
+
 	it := spec.Types.Iterate()
 	for it.Next() {
 		v, ok := it.Type.(*btf.Var)
 		if !ok {
 			continue
 		}
+		if !strings.HasPrefix(v.Name, prefix) {
+			continue
+		}
 
-		if strings.HasPrefix(v.Name, prefix) {
-			return strings.TrimPrefix(v.Name, prefix)
+		err := checkMacroParamVar(v)
+		if err != nil {
+			resultError = multierror.Append(resultError, err)
+			continue
 		}
+		resultNames = append(resultNames, strings.TrimPrefix(v.Name, prefix))
 	}
 
-	return ""
+	return resultNames, resultError
 }
 
 // getTracerMapFromeBPF returns the tracer map from the eBPF object.
 // It looks for maps marked with GADGET_TRACE_MAP() and returns the first one.
 func getTracerMapFromeBPF(spec *ebpf.CollectionSpec) *ebpf.MapSpec {
-	mapName := getGadgetIdentByPrefix(spec, gadgets.TraceMapPrefix)
-	return spec.Maps[mapName]
+	mapNames, _ := getGadgetIdentByPrefix(spec, gadgets.TraceMapPrefix)
+	if len(mapNames) != 1 {
+		log.Warnf("Expected one trace map, found %d", len(mapNames))
```

Comment:
I think this message can be misleading. It's expected that gadgets don't provide a trace map, for instance snapshotters. I'd propose something like:

```golang
// Not tracer map provided
if len(mapNames) == 0 {
return nil;
}

if len(mapNames) > 1 {
log.Warn("multiple tracer maps found, using %q", ....)
}

return spec.Maps[mapNames[0]]
```